core/src/broadcast_stage/standard_broadcast_run.rs

```diff
@@ -331,19 +339,24 @@ impl StandardBroadcastRun {
     ) {
         // Insert shreds into blockstore
         let insert_shreds_start = Instant::now();
-        // The first shred is inserted synchronously
-        let data_shreds = if !shreds.is_empty() && shreds[0].index() == 0 {
-            shreds[1..].to_vec()
-        } else {
-            shreds.to_vec()
-        };
+        let mut shreds = Arc::try_unwrap(shreds).unwrap_or_else(|shreds| (*shreds).clone());
```

Comment:
sneaky little try_unwrap() optimization here 😃 . Should only work if the broadcast finishes with its version of these shreds before we insert here.